File: native/android/aosp-overlay/packages/apps/BuipuiControl/src/org/buipui/control/MainActivity.java
package org.buipui.control;

import android.app.Activity;
import android.os.Bundle;
import android.widget.TextView;

public final class MainActivity extends Activity {
    @Override
    protected void onCreate(Bundle state) {
        super.onCreate(state);

        TextView status = new TextView(this);
        status.setText(
            "Buipui Native Android\n\n" +
            "Control boundary online.\n" +
            "Native core integration is staged for Cuttlefish validation.\n" +
            "Security policy remains AOSP/SELinux authoritative."
        );
        status.setPadding(48, 48, 48, 48);
        setContentView(status);
    }
}

File: mini-os/android/apps/rnd-os/app/src/main/java/com/biupiu/rndos/MainActivity.kt
package com.biupiu.rndos

import androidx.activity.ComponentActivity
import android.os.Bundle
import androidx.activity.compose.setContent

class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContent { BiupiuApp() }
    }
}
